### PR TITLE
fix: show overview and transaction tab on cosmos asset page

### LINKED ISSUE #1291: Overview and Tx tab options not showing on ATOM asset page
### Overview

When selecting ATOM from the asset page, there are is no option for the overview or transaction tabs like there is for other ETH/ERC-20 and BTC assets. When selecting it from the accounts menu however, the option for overview or tx tabs is available.

### References and additional details

1) Open app.shapeshift.com and log into Native wallet.
2) Turn on the cosmos testing flag.
3) From the asset page search for Cosmos/ATOM asset and select.
4) Notice under the asset name there is no option to select overview or transactions.
5) Click on the Accounts page and select Cosmos/ATOM.
6) Notice the option to choose overview or transaction tabs.

image

image

### Acceptance Criteria

Display could be the same for all assets across ATOM/ETH/ERC-20/BTC on Accounts and Asset Page

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Sponsor / Stakeholder

_No response_

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description
- `CosmosAsset.tsx` were missing the subroutes `Overview` and `Transactions`
- I added both routes but as the asset is not wired up right now, we needed to duplicate the `AssetTxHistory` file into the cosmos plugin in order to provide the chainId using a prop
- This file will be removed when it gets wired up because the chainId will be a `param` and we will be able to use both `Asset` and `AssetTxHistory`

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #1291
## Risk
- No risk, behind a feature flag and not wired up so

## Testing
- Go to Cosmos asset page and see if `Overview` and `Transactions` tabs are showing and working as expected

## Screenshots (if applicable)
image
image

### CODE REVIEW

File: src/plugins/cosmos/index.tsx [hunk 1]
```diff
@@ -44,14 +45,38 @@ export function register(): Plugins {
...
+            routes: [
+              {
+                path: '/',
+                label: 'navBar.overview',
+                main: () => <CosmosAsset chainId={'cosmos:osmosis-1'} />
+              },
+              {
+                path: '/transactions',
+                label: 'navBar.transactions',
+                main: () => <CosmosAssetTxHistory chainId={'cosmos:cosmoshub-4'} />
```
Comment (gomesalexandre) [inline src/plugins/cosmos/index.tsx:58]:
```suggestion
                main: () => <CosmosAssetTxHistory chainId={'osmos:osmosis-1'} />
```

**Review by gomesalexandre — CHANGES_REQUESTED** [review]

File: src/plugins/cosmos/index.tsx (hunk 1 above)
Comment (NeOMakinG) [reply src/plugins/cosmos/index.tsx]: My bad, looks like I forgot to test Cosmos on Osmosis transactions tab, it's fixed now 👍

**Review by gomesalexandre — APPROVED** [review]
🐐

File: src/plugins/cosmos/CosmosAsset.tsx [hunk 2]
```diff
@@ -12,10 +12,14 @@ import { useAppDispatch, useAppSelector } from 'state/store'
 
 import { CosmosAssetAccountDetails } from './CosmosAssetAccountDetails'
 
+export interface MatchParams {
```
Comment (0xdef1cafe) [inline src/plugins/cosmos/CosmosAsset.tsx:15]:
```suggestion
export type MatchParams {
```

Comment (0xdef1cafe) [reply src/plugins/cosmos/CosmosAsset.tsx:15]: nitpick - but we only use types. interface should only be used where they define something that is then implemented elsewhere

File: src/plugins/cosmos/CosmostAssetTxHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { useParams } from 'react-router-dom'
+import { AssetHeader } from 'components/AssetHeader/AssetHeader'
+import { Main } from 'components/Layout/Main'
+import { AssetTransactionHistory } from 'components/TransactionHistory/AssetTransactionHistory'
+
+import { MatchParams } from './CosmosAsset'
+
+export const CosmosAssetTxHistory = (props: { chainId: string }) => {
```
Comment (0xdef1cafe) [inline src/plugins/cosmos/CosmostAssetTxHistory.tsx:8]:
```suggestion
export const CosmosAssetTxHistory: React.FC<CosmosAssetTxHistoryProps> = ({ assetSubId, chainId }) => {
```

then we can clean up below

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
nit pick requested changes but otherwise good

File: src/plugins/cosmos/CosmostAssetTxHistory.tsx (hunk 3 above)
Comment (NeOMakinG) [reply src/plugins/cosmos/CosmostAssetTxHistory.tsx]: I don't really get why you added `assetSubId` as a prop, pushed the prop refactor with `chaindId`, tell me if it looks good for you, overwise I missed something!

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**